modules/imgcodecs/src/loadsave.cpp

```diff
@@ -452,8 +452,6 @@ static bool imwrite_( const String& filename, const Mat& image,
     Mat temp;
     const Mat* pimage = &image;
 
-    CV_Assert( image.channels() == 1 || image.channels() == 3 || image.channels() == 4 );
```

Comment:
Yes, I was under the impression encoders had their own check already but I see it's not quite the case, many work under the assumption that `channels > 1` means 3 or 4 channels and don't handle properly the 2 channel case.
I do think it would make sense to remove this assertion here, or check only for `channels > 4` and make each encoder check by themselves if the image has an appropriate number of channels depending on what they support and make the ones that can support grayscale+alpha do support it since it's a totally valid use case I think.
I'll make this PR purely about binary transparency decoding not to overburden it and make another one that attempts to add proper 2-channel images handling to all encoders.